modules/formulize/templates/admin/element_names.html

```diff
@@ -80,12 +80,16 @@
 
 <script type="text/javascript">
 $(".savebutton").click(function() {
-  if($("[name=elements-ele_caption]").val() == "") {
+	if($("[name=elements-ele_caption]").val() == "") {
 		alert("Elements must have a caption!");
 		$("[name=elements-ele_caption]").focus();
 	}
 	<{* reload after ele_handle is blank, since it will be reset to the ele_id and that's a state change we'll need to reload to show the user *}>
 	if($("[name=elements-ele_handle]").val() == "") {
+		if ("<{$content.ele_id}>" == "new") {
+			var str="<{$content.formhandle}>"+"_"+$("[name=elements-ele_caption]").val().replace(new RegExp(" ","gm"),"_");

```

Comment:
This does replace spaces with _, but it should also make the text all lower-case and remove any punctation, basically, remove anything not A-Z or a-z or space.